contracts/NodeDelegator.sol

```diff
@@ -291,6 +318,36 @@ contract NodeDelegator is INodeDelegator, LRTConfigRoleChecker, PausableUpgradea
         );
     }
 
+    /// @notice Undelegates all staked assets from this NodeDelegator from the
+    /// previously delegated to EigenLayer Operator.
+    /// This also forces a withdrawal so the assets will need to be claimed.
+    function undelegate() external onlyLRTManager {
+        address delegationManagerAddress = lrtConfig.getContract(LRTConstants.EIGEN_DELEGATION_MANAGER);
+        IDelegationManager delegationManager = IDelegationManager(delegationManagerAddress);
+
+        // Get the amount of strategy shares owned by this NodeDelegator contract
+        IStrategyManager strategyManager = IStrategyManager(lrtConfig.getContract(LRTConstants.EIGEN_STRATEGY_MANAGER));
+
+        // For each asset
+        address[] memory assets = lrtConfig.getSupportedAssetList();
+        for (uint256 i = 0; i < assets.length;) {
+            // Get the EigenLayer strategy for the asset
+            // The WETH asset will point to the EigenLayer beaconChainETHStrategy
+            // 0xbeaC0eeEeeeeEEeEeEEEEeeEEeEeeeEeeEEBEaC0
+            address strategy = lrtConfig.assetStrategy(assets[i]);
+
+            // account for the strategy shares pending internal withdrawal
+            pendingInternalShareWithdrawals[strategy] +=
+                strategyManager.stakerStrategyShares(address(this), IStrategy(strategy));
+
+            unchecked {
```

Comment:
Let's just put this ++I where it belongs in the for, lol. We are wayyyyyy off the hot path here.

Reply:
I've removed the unchecked.

BTW, PrimeETH is using Solidity 0.8.21. Solidity 0.8.22 Unchecked Loop Increments https://docs.soliditylang.org/en/latest/internals/optimizer.html#unchecked-loop-increment

We could upgrade a minor version and remove the unchecked increments in loops.